cli_tools/gce_vm_image_import/importer/api_inflater.go

```diff
@@ -87,8 +95,22 @@ func (inflater *apiInflater) inflate(loggableBuilder *service.SingleImageImportL
 		return persistentDisk{}, inflationInfo{}, err
 	}
 
-	// Cleanup the shadow disk ignoring error
-	defer inflater.computeClient.DeleteDisk(inflater.args.Project, inflater.args.Zone, cd.Name)
+	defer func() {
+		// Cleanup the shadow disk ignoring error
+		inflater.computeClient.DeleteDisk(inflater.args.Project, inflater.args.Zone, cd.Name)
+
+		// Unblock the "cancel" function by receiving the msg.
+		select {
+		case <-inflater.cancelChan:
+		default:
+		}
+	}()
+
+	select {
+	case <-inflater.cancelChan:
+		return persistentDisk{}, inflationInfo{}, nil
+	default:
+	}
```

Comment:
Why is listening to cancelChan done multiple times in this func?

Reply:
I am doing unblocking receiving in multiple places in order to "cancel" as soon as possible.

Reply:
Simplified, only kept this one, since this is the only important timing to return early.